### PR TITLE
Update analyzers and frameworks to latest versions

### PR DESCRIPTION
**Pull request by pakdev**
# Justification
ASW now [supports](https://ni.visualstudio.com/DevCentral/_git/ASW/commit/238d18d6c98bab350a66e989570c7b3aad07f5f6?refName=refs%2Fheads%2Fmain&tab=details) VS2019 16.9+ ([decision](https://niweb.natinst.com/confluence/display/LV/Everything.sln:+Drop+support+for+VS2017+in+favor+of+VS2019)). This [allows](https://docs.microsoft.com/en-us/visualstudio/extensibility/roslyn-version-support?view=vs-2019#:~:text=.NET%20compiler%20platform%20package%20version%20reference%20%20,2017%20version%2015.7%20%2011%20more%20rows%20) us to use the latest versions of Roslyn analyzers.

# Implementation
- Updated non-analyzer frameworks to be NET Core 5 or NET 4.6.1 (required for a Windows namespace)
- Updated analyzer frameworks to be NET Standard 2.0 (framework used when creating an analyzer through VS template)
- Updated analyzer references to be the latest stable version (except for StyleCop, which has not moved out of beta for years)
- Added project-specific PackageReferences

# Testing
- Ensured code builds and tests pass

+semver: breaking

### CODE REVIEW

**Comment by jryckman** [comment]
I'm adding @ccifra and @gregr-ni  to this to get their thoughts on the update. This would still need to be a 2.0 release of the analyzer since it's a big change.

**Comment by gregr-ni** [comment]
> I'm adding @ccifra and @gregr-ni to this to get their thoughts on the update. This would still need to be a 2.0 release of the analyzer since it's a big change.

We do need to stay up to date on these. Does seem like a 2.0 is appropriate as this is likely to require work on clients to adopt.

File: Directory.Build.props [hunk 1]
```diff
@@ -62,12 +62,11 @@
     </PropertyGroup>
 
     <ItemGroup>
-        <PackageReference Include="Microsoft.CodeAnalysis" Version="2.10.0" />
-        <PackageReference Include="Microsoft.CodeAnalysis.CSharp" Version="2.10.0" />
-        <PackageReference Include="Microsoft.CodeAnalysis.FxCopAnalyzers" Version="2.9.10" />
-        <PackageReference Include="Microsoft.VisualStudio.Threading.Analyzers" Version="16.0.102" />
-        <PackageReference Include="StyleCop.Analyzers" Version="1.1.118" />
-        <PackageReference Include="Newtonsoft.Json" Version="12.0.1" />
```
Comment (jryckman) [inline Directory.Build.props:66]: Do we not need Newtonsoft.Json anymore?

Comment (pakdev) [reply Directory.Build.props:66]: Build and tests still work without it 🤷

Comment (pakdev) [reply Directory.Build.props:66]: Well, at least locally. Looks like updating the analyzers has imposed new rules that must be corrected for the CI to pass

**Comment by jryckman** [comment]
@pakdev btw I'm just waiting for all the checks to pass and I'll take a look again this. No rush, just wanted to let you know

**Comment by pakdev** [comment]
> @pakdev btw I'm just waiting for all the checks to pass and I'll take a look again this. No rush, just wanted to let you know

Yeah, got busy. Hopefully the build is fixed now

**Comment by pakdev** [comment]
@jryckman this is ready for review

**Review by gregr-ni — APPROVED** [review]

**Comment by jryckman** [comment]
@pakdev since this is a big change is will result in the analyzer doing a major version bump, how is that handled? If we don't make any modifications outside of this PR, it will just autobump the version and publish a newer minor version, right?

**Comment by Everest2020** [comment]
@jryckman Yes, I think so. Since I enabled `commmit-message-incrementing`, one can use `+semver: breaking` or `+semver: major` to increase the major version.

You can also directly modify GitVersion.yml by setting up the `next-version`. 

Details can be found https://gitversion.net/docs/reference/version-increments

**Comment by pakdev** [comment]
Ok, I added `+semver: breaking` at the end of my commit message. That's sufficient, right?

**Comment by jryckman** [comment]
> 
> 
> Ok, I added `+semver: breaking` at the end of my commit message. That's sufficient, right?

I think so? I haven't used that before, so I'm not sure what's going to happen. @Everest2020 would know best as he's the one who set it all up

**Comment by Everest2020** [comment]
@pakdev I disabled the step to publish the Nuget package. Feel free to commit this PR, I will take a look at whether the generated Nuget version is as expected tomorrow **after this commit is in**.

**Comment by jryckman** [comment]
> 
> 
> @pakdev I disabled the step to publish the Nuget package. Feel free to commit this PR, I will take a look at whether the generated Nuget version is as expected tomorrow **after this commit is in**.

Thanks @Everest2020 !

**Review by jryckman — APPROVED** [review]

**Pull request merged**